src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+            var destinationSyntaxNode = await codeGenerationService.FindMostRelevantNameSpaceOrTypeDeclarationAsync(
+                document.Project.Solution, result.Destination, default, cancellationToken).ConfigureAwait(false);
+            var (editorMap, syntaxMap) = await InitializeEditorMapsAndSyntaxMapAsync(result, solution, solutionEditor, destinationSyntaxNode, cancellationToken).ConfigureAwait(false);
+            // Add members to destination
+            var pullUpMembersSymbols = result.MemberAnalysisResults.SelectAsArray(
+                memberResult =>
+                {
+                    if (memberResult.MakeDeclarationAtDestinationAbstract)
+                    {
+                        // Change the member to abstract if user choose to make them abstract
+                        return GetAbstractMemberSymbol(memberResult.Member);
+                    }
+                    else
+                    {
+                        return memberResult.Member;
+                    }
+                });
+            var options = new CodeGenerationOptions(reuseSyntax: true, generateMethodBodies: false);
+            var newDestination = codeGenerationService.AddMembers(destinationSyntaxNode, pullUpMembersSymbols, options: options);
+
+            // Remove some original members since we are pulling members into class.
+            // Note: If user chooses to make the member abstract, then the original member won't be touched,
+            // It will just pull abstract declaration up to destination.
+            // But if the member is abstract itself, it will still be removed.
+            foreach (var analysisResult in result.MemberAnalysisResults)
+            {
+                if (!analysisResult.MakeDeclarationAtDestinationAbstract)
+                {
+                    foreach (var syntax in syntaxMap[analysisResult.Member])
+                    {
+                        var originalMemberEditor = editorMap[syntax.SyntaxTree];
+                        originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
+                    }
+                }
+            }
+
+            // Change the destination to abstract class if needed.
+            var destinationEditor = editorMap[destinationSyntaxNode.SyntaxTree];
+            if (!result.Destination.IsAbstract &&
+                result.MemberAnalysisResults.Any(analysis => analysis.Member.IsAbstract || analysis.MakeDeclarationAtDestinationAbstract))
+            {
+                var modifiers = DeclarationModifiers.From(result.Destination).WithIsAbstract(true);
+                newDestination = destinationEditor.Generator.WithModifiers(newDestination, modifiers);
+            }
+
+            destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private ISymbol GetAbstractMemberSymbol(ISymbol member)
```

Comment:
Consider MakeAbstractVersion.  Saying 'Get' makes it seem as if you're going and looking for this.